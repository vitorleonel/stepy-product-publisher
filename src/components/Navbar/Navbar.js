import Button from '../Button';

const Navbar = () => {
  return (
    <header className="navbar">
      <div className="navbar-brand">
        <img src="/images/brand.svg" alt="Stepy" />
        <h1>Product Publisher</h1>
      </div>

      <nav className="navbar-menu">
        <a
          href="https://github.com/vitorleonel/stepy-product-publisher"
          target="_blank"
          rel="noopener noreferrer"
        >
          <Button text="GitHub Repository" leftIcon="logo-github" />
        </a>
      </nav>
    </header>
  );
};

export default Navbar;
