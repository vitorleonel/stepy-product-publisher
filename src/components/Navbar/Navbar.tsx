const Navbar = (): JSX.Element => {
  return (
    <header className="navbar">
      <div className="navbar-brand">
        <img src="/images/brand.svg" alt="Stepy" />
        <h1>Product Publisher</h1>
      </div>

      <nav className="navbar-menu">
        <a href="https://github.com/vitorleonel/stepy-product-publisher" target="_blank" rel="noopener noreferrer">
          <i className="fab fa-github"></i>
          <span>Github Repository</span>
        </a>
      </nav>
    </header>
  );
}

export default Navbar;
