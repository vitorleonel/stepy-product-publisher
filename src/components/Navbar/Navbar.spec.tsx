import { render } from '@testing-library/react';

import Navbar from './Navbar';

describe('Navbar Component', () => {
  test('should render correctly', () => {
    const { container } = render(<Navbar />);

    expect(container).toMatchSnapshot();
  });
});
