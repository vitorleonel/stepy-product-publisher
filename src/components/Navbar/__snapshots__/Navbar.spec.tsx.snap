// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component should render correctly 1`] = `
<div>
  <div
    class="bp3-navbar"
  >
    <div
      class="bp3-navbar-heading"
    >
      <img
        alt="Stepy"
        src="/images/brand.svg"
      />
      <h1>
        Product Publisher
      </h1>
    </div>
    <a
      href="https://github.com/vitorleonel/stepy-product-publisher"
      rel="noopener noreferrer"
      target="_blank"
    >
      <button
        class="bp3-button bp3-large bp3-minimal"
        type="button"
      >
        <span
          class="bp3-icon bp3-icon-git-repo"
          icon="git-repo"
        >
          <svg
            data-icon="git-repo"
            height="16"
            viewBox="0 0 16 16"
            width="16"
          >
            <desc>
              git-repo
            </desc>
            <path
              d="M5 9H4v1h1V9zm10-9H1C.45 0 0 .45 0 1v13c0 .55.45 1 1 1h3v1l2-1 2 1v-1h7c.55 0 1-.45 1-1V1c0-.55-.45-1-1-1zM4 13H2v-1h2v1zm10 0H8v-1h6v1zm0-2H3V2h11v9zM5 3H4v1h1V3zm0 4H4v1h1V7zm0-2H4v1h1V5z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <span
          class="bp3-button-text"
        >
          GitHub Repository
        </span>
      </button>
    </a>
  </div>
</div>
`;
